src/sentry/workflow_engine/endpoints/validators/base/workflow.py

```diff
@@ -1,22 +1,103 @@
     triggers = BaseDataConditionGroupValidator(required=False)
-    action_filters = serializers.ListField(child=BaseDataConditionGroupValidator(), required=False)
+    action_filters = serializers.ListField(required=False)
 
     # TODO - Need to improve the following fields (validate them in db)
     organization_id = serializers.IntegerField(required=True)
     environment_id = serializers.IntegerField(required=False)
 
+    def _split_action_and_condition_group(
+        self, action_filter: dict[str, Any]
+    ) -> tuple[ActionData, DataConditionGroupData]:
+        try:
+            actions = action_filter["actions"]
+        except KeyError:
+            raise serializers.ValidationError("Missing actions key in action filter")
+
+        return actions, action_filter
+
     def validate_config(self, value):
         schema = Workflow.config_schema
         return validate_json_schema(value, schema)
+
+    def validate_action_filters(self, value):
+        for action_filter in value:
+            actions, condition_group = self._split_action_and_condition_group(action_filter)
+            BaseDataConditionGroupValidator(data=condition_group).is_valid(raise_exception=True)
+
+            for action in actions:
+                BaseActionValidator(data=action).is_valid(raise_exception=True)
+
+        return value
+
+    def create(self, validated_value: dict[str, Any]) -> Workflow:
+        condition_group_validator = BaseDataConditionGroupValidator()
+        action_validator = BaseActionValidator()
+
+        with transaction.atomic(router.db_for_write(Workflow)):
+            when_condition_group = condition_group_validator.create(validated_value["triggers"])
+
+            workflow = Workflow.objects.create(
+                name=validated_value["name"],
+                enabled=validated_value["enabled"],
+                config=validated_value["config"],
+                organization_id=validated_value["organization_id"],
+                environment_id=validated_value.get("environment_id"),
+                when_condition_group=when_condition_group,
+            )
+
+            # Create actions and action filters, then associate them to the workflow
+            for action_filter in validated_value["action_filters"]:
+                actions, condition_group = self._split_action_and_condition_group(action_filter)
+                new_condition_group = condition_group_validator.create(condition_group)
+
+                # Connect the condition group to the workflow
+                WorkflowDataConditionGroup.objects.create(
+                    condition_group=new_condition_group,
+                    workflow=workflow,
+                )
+
+                for action in actions:
+                    new_action = action_validator.create(action)
+                    DataConditionGroupAction.objects.create(
+                        action=new_action,
+                        condition_group=new_condition_group,
+                    )
```

Comment:
super nit: can maybe bulk create the `WorkflowDataConditionGroups` and `DataConditionGroupActions` but idk how much time that would actually save